src/Evolution/Systems/GeneralizedHarmonic/UpwindPenaltyCorrection.hpp

```diff
@@ -0,0 +1,236 @@
+ * implemented since that boundary term does not consistently treat both sides
+ * of the interface on the same footing.
+ *
+ * For the first-order GH system the correction is:
+ *
+ * \f{align}{
+ *   \label{eq:pnpm upwind gh metric}
+ *   D_{g_{ab}} &= \tilde{\lambda}_{v^{g}}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},g}_{ab}
+ *                - \tilde{\lambda}_{v^{g}}^{\mathrm{int}}
+ *                v^{\mathrm{int},g}_{ab}, \\
+ *   \label{eq:pnpm upwind gh pi}
+ *   D_{\Pi_{ab}}
+ *              &= \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},+}_{ab} +
+ *                \tilde{\lambda}_{v^-}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},-}_{ab}\right)
+ *                + \tilde{\lambda}_{v^g}^\mathrm{ext}\gamma_2
+ *                v^{\mathrm{ext},g}_{ab}
+ *                \notag \\
+ *              &-\frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{int}}
+ *                v^{\mathrm{int},+}_{ab} +
+ *                \tilde{\lambda}_{v^-}^{\mathrm{int}}
+ *                v^{\mathrm{int},-}_{ab}\right)
+ *                - \tilde{\lambda}_{v^g}^\mathrm{int}\gamma_2
+ *                v^{\mathrm{int},g}_{ab} , \\
+ *   \label{eq:pnpm upwind gh phi}
+ *   D_{\Phi_{iab}}
+ *              &= \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},+}_{ab}
+ *                - \tilde{\lambda}_{v^-}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},-}_{ab}\right)n_i^{\mathrm{ext}}
+ *                + \tilde{\lambda}_{v^0}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},0}_{iab}
+ *                \notag \\
+ *              &-
+ *                \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{int}}
+ *                v^{\mathrm{int},+}_{ab}
+ *                - \tilde{\lambda}_{v^-}^{\mathrm{int}}
+ *                v^{\mathrm{int},-}_{ab}\right)n_i^{\mathrm{int}}
+ *                - \tilde{\lambda}_{v^0}^{\mathrm{int}}
+ *                v^{\mathrm{int},0}_{iab},
+ * \f}
+ *
+ * with characteristic fields
+ *
+ * \f{align}{
+ *   \label{eq:pnpm gh v g}
+ *   v^{g}_{ab} &= g_{ab}, \\
+ *   \label{eq:pnpm gh v 0}
+ *   v^{0}_{iab} &= (\delta^k_i-n^k n_i)\Phi_{kab}, \\
+ *   \label{eq:pnpm gh v pm}
+ *   v^{\pm}_{ab} &= \Pi_{ab}\pm n^i\Phi_{iab} -\gamma_2 g_{ab},
+ * \f}
+ *
+ * and characteristic speeds
+ *
+ * \f{align}{
+ *   \lambda_{v^g} =& -(1+\gamma_1)\beta^i n_i -v^i_g n_i, \\
+ *   \lambda_{v^0} =& -\beta^i n_i -v^i_g n_i, \\
+ *   \lambda_{v^\pm} =& \pm \alpha - \beta^i n_i - v^i_g n_i,
+ * \f}
+ *
+ * where \f$v_g^i\f$ is the mesh velocity. We have also defined
+ *
+ * \f{align}{
+ *   \tilde{\lambda}_{\hat{\alpha}} =
+ *    \left\{
+ *        \begin{array}{ll}
+ *          \lambda_{\hat{\alpha}} &
+ *            \mathrm{if}\;\lambda_{\hat{\alpha}}\le 0.0 \\
+ *          0 & \mathrm{otherwise}
+ *        \end{array}\right.
+ * \f}
+ *
+ * Note that we have assumed \f$n_i^{\mathrm{ext}}\f$ points in the same
```

Comment:
Clarify that here we assume that the normals point in the same direction, but in the code, they point in opposite directions. It would be even better to add to the help text the equations with the spectre, sign-flipping normals explicitly, so you can compare to exactly what's coded up.